### DISCUSSION TITLE
Dont generate code for / ignore certain methods?

### DISCUSSION THREAD

**Discussion by deliciouslytyped** [mapstruct/mapstruct#3507] [Q&A]
I have some methods in my mappers / I'm doing some extra work where those methods can't be generated by mapstruct, and that need to be implemented by the developer. Is there any way to tell mapstruct ignore certain methods? I've been searching and haven't found a way to do this.

Specifically, I have mappers that implement another interface, and it would be great if I could somehow have mapstruct ignore the interface's methods.

Edit:
Of course, all one has to do is ask and then they start finding solutions... Apparently it works to add default implementations on the interface, but that's very ugly. Is there any better solution?

**Comment by thunderhook** [comment]
Hi @deliciouslytyped!

There is currently no such option. However, I have found two issues that relate to it:
* #658 
* (somewhat related) #101 

Since MapStruct supports both interfaces and abstract classes annotated with `@Mapper`, this is a bit more complex than it looks at first. Details about the complexity can be found here: https://github.com/mapstruct/mapstruct/issues/658#issuecomment-147825481

However, you can add a thumbs up emoji in issue #658. This feature request is about 9 years old, so I wouldn't count on it being implemented in the near future.
